equinox/nn/_embedding.py

```diff
@@ -89,3 +96,313 @@ def __call__(
                 "`eqx.nn.Embedding()(x)` should be called with a scalar index `x`. "
                 "Use `jax.vmap` if you would like to index with multiple values."
             )
+
+
+class RotaryPositionalEmbedding(Module, strict=True):
+    """
+    A rotary positional encoding module, as described in the paper
+    "RoFormer: Enhanced Transformer with Rotary Position Embedding". While
+    this module can be used in any context, it is particularly useful for
+    providing positional information to transformer models.
+
+    !!! example
+        The following example demonstrates how to use `RotaryPositionalEmbedding` in
+        a simple transformer model.
+        ```python
+
+        class TransformerBlock(eqx.nn.StatefulLayer):
+            ...
+            key_rope_embeddings: RotaryPositionalEmbedding
+            query_rope_embeddings: RotaryPositionalEmbedding
+
+            def __init__(...):
+                ...
+                self.query_rope_embeddings = RotaryPositionalEmbedding(
+                    embedding_size=n_embd
+                )
+                self.key_rope_embeddings = RotaryPositionalEmbedding(
+                    embedding_size=n_embd
+                )
```

Comment:
I think we should be able to use just a single `RotaryPositionalEmbedding` layer?